wordpress-mobile/WordPress-iOS
# [Gutenberg] Implement "Other Apps" media source option

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/1265

This PR adds the option to select media items from the iOS Files App via "Other Apps" option.

This works for all media related blocks, and the user should be allowed to pick just the corresponding media types for the blocks.

For this to work, the `filters` property was needed from JS, but it wasn't given via `requestOtherMediaPick` methods. The `gutenberg` side changes are meant to allow this property to be passed on any `requestMediaPick` for any source.

`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1547
`gutenberg` PR: https://github.com/WordPress/gutenberg/pull/18303

![other_apps](https://user-images.githubusercontent.com/9772967/68289781-115b4300-0087-11ea-9444-bf5958366a6d.gif)

**Note:**
- As shown in the GIF, there seems to be a bug on iOS 13 where the bar buttons on the Documents Picker are not visible. This is also the case for Aztec and WPMediaLibrary instances.
I have decided to not fix this on this particular PR.

**To Test:**

- Checkout [`gutenberg-mobile` PR]( https://github.com/wordpress-mobile/gutenberg-mobile/pull/1547) and **Run Metro server.**

Image Block:
- Add an Image block.
- Check that the "Other Apps" option is there.
- Open the "Other Apps" picker.
- Check that only images are selectable.
- Choose one image.
- Check that the selected image is added to the block.
- Smoke-test other source options.

Video Block:
- Add a Video block.
- Check that the "Other Apps" option is there.
- Open the "Other Apps" picker.
- Check that only videos are selectable.
- Choose one video.
- Check that the selected video is added to the block.
- Smoke-test other source options.

Media&Text:
- Add an Media&Text block.
- Check that the "Other Apps" option is there.
- Open the "Other Apps" picker.
- Check that you can select only images and videos.
- Choose one image or video.
- Check that the image or video selected was added.
- Smoke-test other source options.

Update release notes:
